api/db-cdr/generate-cdr/libproject/genomic_manifests.rb

```diff
@@ -385,37 +436,66 @@ def build_manifests_for_aw4_section(input_section, ingest_bucket, dest_bucket, d
     output_manifest = []
   end
 
+  prev_manifest = _read_previous_manifest(
+    project,
+    dest_bucket,
+    input_section["deltaReleaseManifestPath"],
+    input_section["deltaRelease"],
+    input_section["pooledDestPathInfix"])
+
   copy_manifest = aw4_rows.flat_map do |aw4_entry|
     unless output_manifest.nil?
-      out_row = {"person_id" => aw4_entry["research_id"]}
-      input_section["outputManifestSpec"].each do |aw4_col, out_col|
-        row = _build_copy_manifest_row(aw4_entry[aw4_col], ingest_base_path, destination, input_section, aw4_entry["research_id"])
-        destination_uri = File.join(row[:destination_dir], File.basename(row[:ingest_path]))
-        out_row[out_col] = destination_uri
+      out_row = { "person_id" => aw4_entry["research_id"] }
+      # If the research_id already exists in the prev manifest, just put as is in the new output manifest
+      unless prev_manifest.nil? or prev_manifest[aw4_entry["research_id"]].nil?
+        output_manifest.push(prev_manifest[aw4_entry["research_id"]].to_h)
+      else
+        input_section["outputManifestSpec"].each do |aw4_col, out_col|
+          row = _build_copy_manifest_row(aw4_entry[aw4_col], ingest_base_path, destination, input_section, aw4_entry["research_id"])
+          destination_uri = File.join(row[:destination_dir], File.basename(row[:ingest_path]))
+          out_row[out_col] = destination_uri
+        end
+        output_manifest.push(out_row)
       end
-      output_manifest.push(out_row)
     end
 
     source_paths = input_section["aw4Columns"].map { |k| aw4_entry[k] }
     source_paths.map do |source_path|
-      _build_copy_manifest_row(source_path, ingest_base_path, destination, input_section, aw4_entry["research_id"])
+      if prev_manifest.nil? or prev_manifest[aw4_entry["research_id"]].nil?
+        _build_copy_manifest_row(source_path, ingest_base_path, destination, input_section, aw4_entry["research_id"])
+      end
     end
   end
 
+  # To remove nils
+  copy_manifest = copy_manifest.compact
+
   unless output_manifest.nil?
     unpooled_prefix = "#{display_version_id}/#{input_section["pooledDestPathInfix"]}"
     unpooled_ingest_base_path = File.join(ingest_bucket, unpooled_prefix)
     unpooled_destination = File.join(dest_bucket, unpooled_prefix)
     copy_manifest.push(_build_copy_manifest_row(
                          output_manifest_path, unpooled_ingest_base_path, unpooled_destination, {
-                           "filenameMatch" => File.basename(output_manifest_path),
-                           "filenameReplace" => "manifest.csv",
-                         }))
+                         "filenameMatch" => File.basename(output_manifest_path),
+                         "filenameReplace" => "manifest.csv",
+                       }))
   end
 
   return [copy_manifest, output_manifest]
 end
 
+def _get_pooled_path(pathInfix, display_version_id, is_delta_release)
```

Comment:
if deltaRelease field is specified, then the directory name will by postfixed by "_delta", otherwise the release is a "_base" one